File: core/fetch/markdown_fetcher.py
from __future__ import annotations

from core.fetch.http_fetcher import fetch_sync, fetch_async


def fetch_markdown_sync(url: str):
    data = fetch_sync(url)
    data["source"] = "markdown"
    return data


async def fetch_markdown_async(url: str):
    data = await fetch_async(url)
    data["source"] = "markdown"
    return data

File: src/fetch/markdownFetcher.ts
/**
 * Converted from Python: core/fetch/markdown_fetcher.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { fetchSync, fetchAsync } from "./httpFetcher.js";

export function fetchMarkdownSync(url: any): any {
  var data: any = fetchSync(url);
  py.setItem(data, "source", "markdown");
  return data;
}
export async function fetchMarkdownAsync(url: any): Promise<any> {
  var data: any = await fetchAsync(url);
  py.setItem(data, "source", "markdown");
  return data;
}
export { fetchAsync, fetchSync };
